app/components/documents/document-preview-sidebar.js

```diff
@@ -23,13 +39,88 @@ export default class DocumentsDocumentPreviewSidebar extends Component {
 
   @task
   *loadData() {
-    const docContainer = yield this.args.piece.documentContainer;
-    this.documentType = yield docContainer.type;
+    this.docContainer = yield this.args.piece.documentContainer;
+    this.loadVersionHistory.perform();
+    this.documentType = yield this.docContainer.type;
```

Comment:
Hier wordt al behoorlijk wat data geladen ongeacht de tab die de gebruiker aan het bekijken is. Als de details tab bijvoorbeeld geopend is, zal toch ook al de hele versie historiek gedownload worden.

Het is waarschijnlijk beter om met aparte componenten te werken voor de inhoud van iedere tab. Op die manier gaat de `document-preview-sidebar` component ook eenvoudiger worden en enkel de logica van de tab-selectie bevatten. Nieuwe components kunnen misschien best samengezet worden in een subfolder `documents/document-preview/details`, `documents/document-preview/versions`, ...

Reply:
Ik zal het details tab component doen in 2704